Visual Display of User Details #144

lposo opened the pull request:
Added to the user page a visual representation of the user's details. Additionally, users can press the "Edit Profile" button to bring up a form to change these details. They can also change their password by clicking the "Change Password" button. There is no implementation of a connection to the backend yet. Additionally, this feature does not completely match the corresponding mock-up due to the fact that the type of settings that a user is to edit has not yet been confirmed.

Note: I pulled from the branch for issue# 139 as that branch had already implemented the user profile page. Thus, I'll wait to commit this until after that branch has been merged.

User Detail Page:
![image](https://user-images.githubusercontent.com/46732530/103114729-d5d65c00-462d-11eb-8f3b-c4d48298ff89.png)

Form to Edit Profile:
![image](https://user-images.githubusercontent.com/46732530/103114744-e686d200-462d-11eb-8973-803c7e24debe.png)

Form to Edit Password
![image](https://user-images.githubusercontent.com/46732530/103114765-f8687500-462d-11eb-84dd-f3d664c66545.png)

lposo committed:
Merge remote-tracking branch 'origin/frontend/visual-table-of-saved-datasets-graphs-#139' into frontend/visual-display-of-user-detail-and-settings-144

lposo committed:
Set up profile page with ability to change profile details and password

Users can now edit their profile information and passwor.d Currently, no backend implementation has been added. Uses dummy data for now.

lposo committed:
Merge remote-tracking branch 'origin/frontend/visual-table-of-saved-datasets-graphs-#139' into frontend/visual-display-of-user-detail-and-settings-144

lposo committed:
Fixed indentation

lposo committed:
Merge branch 'develop' into frontend/visual-display-of-user-detail-and-settings-144

lposo committed:
Added validation for user detail form.

Now checks if all parts are filled out, as well as if the email entered is a valid email.

lposo committed:
empty commit

ds161833 committed:
Merge branch 'Frontend/lab-form-refactor' into Frontend1/visual-display-of-user-detail-and-settings-144

ds161833 committed:
Sandbox

lposo committed:
Cleaned up Profile forms and password forms.

Implemented forms in a similar manner to the form for entering dataset information. Users can toggle between changing their profile settings and changing their password. Currently not connected to the backend. Additionally, does not match the mockups due to I not knowing of any other settings that need to be implemented.

lposo committed:
Made forms disappear upon submitting the forms.

lposo committed:
Merge branch 'develop' into Frontend1/visual-display-of-user-detail-and-settings-144

lposo committed:
Merge branch 'develop' into Frontend1/visual-display-of-user-detail-and-settings-144

JeremiahTee commented:
As mentioned during the Scrum on Monday 28th,
Waiting on @laithAwad 's PR to be merged since @lposo imported some of his code.

JeremiahTee commented:
@lposo Started work on this more than 9 days ago but forgot to move the card to "Progress". Reminder to all to move their cards in Zenhub to make sure the progress is being tracked correctly 💯

GD-Dheer commented on Client/src/Components/Profile/PasswordForm.tsx:
```diff
@@ -0,0 +1,43 @@
+  const validationSchema = Yup.object().shape({
+    password: Yup.string().required('Password is required').min(8, 'Password should have a minimum of 8 characters'),
+    passwordConfirmation: Yup.string()
+      .oneOf([Yup.ref('password'), null], 'Passwords must match')
```
Nice use of yup & formik

GD-Dheer commented on Client/src/Components/Profile/PasswordSection/PasswordSection.tsx:
```diff
@@ -0,0 +1,26 @@
+import { classStyles } from '../../../appTheme'
+
+interface IProps {
+
```
Empty Iprops?

GD-Dheer commented on Client/src/Components/Profile/ProfileForm.tsx:
```diff
@@ -0,0 +1,51 @@
+  const { user, onSubmit } = props
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string().required(),
```
Do we need this validation schema? We shouldnt be allowing users to change name, DOB.

GD-Dheer commented on Client/src/Components/Profile/PasswordSection/PasswordSection.tsx:
```diff
@@ -0,0 +1,26 @@
+import { Box, Grid, Typography } from '@material-ui/core'
```
Can you rename this file? Profile/PasswordSection/PasswordSection is vague

GD-Dheer commented on Client/src/Components/Profile/ProfileSection/ProfileSection.tsx:
```diff
@@ -0,0 +1,32 @@
+import { Box, Grid, Typography } from '@material-ui/core'
```
Can we rename this one too

GD-Dheer commented on Client/src/Components/Profile/ProfileSection/ProfileSection.tsx:
```diff
@@ -0,0 +1,32 @@
+import { classStyles } from '../../../appTheme'
+
+interface IProps {
+
```
Empty?

GD-Dheer commented on Client/src/Views/ProfileView.tsx:
```diff
@@ -0,0 +1,225 @@
+}
+
+// eslint-disable-next-line no-undef
+function Row(props: { row: ReturnType<typeof createData> }) {
```
Does it make more sense to move this into a separate component?

lposo replied:
Maybe, but that section is from @laithAwad 's branch, so I didn't want to touch anything.

lposo replied:
After talking with @laithAwad , we decided to leave it as is. It would take more time to move it which could be used to finish up other stories, so for now the function will stay like this,

GD-Dheer commented on Client/src/Components/Profile/ProfileForm.tsx:
```diff
@@ -0,0 +1,51 @@
+
+export interface IPasswordSettings {
+  password: string,
+  passwordConfirmation: string
```
is this required here?

GD-Dheer replied:
We have the same interfaces in multiple places, maybe move them in a common place?

GD-Dheer commented on Client/src/Components/Profile/ProfileSection/ProfileDetails.tsx:
```diff
@@ -0,0 +1,63 @@
+      </Grid>
+    </Grid>
+  )
+}
```
Can we maybe improve the Ui a bit here? This can be helpful maybe https://www.behance.net/

GD-Dheer reviewed (changes requested):
Left some feedback. Thanks for the work! Also can we use tabs to clean up the UI? One for saved datasets and another for profile related stuff. Material offers this feature.

GD-Dheer commented:
> Left some feedback. Thanks for the work! Also can we use tabs to clean up the UI? One for saved datasets and another for profile related stuff. Material offers this feature.

Also we should not allow Users to change their Date of Birth, first or last name, email maybe but general idea for stronger security is to just have a user create a new account. We give admin access to accounts based on emails.

Users should only be able to change organization & password. @lposo @TiraFire

lposo committed:
Renamed files and made additional changes based on feedback.

File names have been changed to make it more clear which file is which. Additionally, some empty props have been removed, and certain interfaces, such as IUser and IPassword settings, have been moved to their own files. Users can no longer change their date of birth or name.

lposo commented:
I made some changes based on what you said @GD-Dheer . I'll try to finish up the UI changes you wanted for the ProfileDetails file by the end of today.

lposo committed:
Slight changes to the user detail design.

Changed user details to use a table instead of a grid. Removed the box around it and used a simple border similar to other ocmponents.

lposo committed:
Merge branch 'develop' into Frontend1/visual-display-of-user-detail-and-settings-144

lposo committed:
Merge branch 'develop' into Frontend1/visual-display-of-user-detail-and-settings-144

lposo committed:
Fixed import of ProfileView in the Navigation Menu.

Added brackets as ProfileView does not have a default export.

lposo committed:
Added tabs to ProfileView

Users can now toggle between seeing user details and seeing favorite details.

GD-Dheer reviewed (approved):
All comments addressed! Well renamed and tabs clean up the UI

silent-po2 reviewed (approved):
With the new changes, this is cleaner, and it looks solid.
